### PR TITLE
VZ-4619 vpo install logging changes.

### PR DESCRIPTION
**Pull request by pfmackin**
- Create controller resource specific logging init function
- Fix some issues with logging related to queued up Verrazzano CRs that are delivered after reconciliation is done.  
- Fix the log initialization at the component level, change componet.For to Init to make it clear that it is an initialization function that must be called for when getting a component level component context.  
- Fine tuning logging messages to comply with some of the bugs entered.
- Add more comments
- Add more tests

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: pkg/helm/helmcli.go [hunk 1]
```diff
@@ -168,22 +169,21 @@ func runHelm(log *zap.SugaredLogger, releaseName string, namespace string, chart
 
 		// mask sensitive data before logging
 		cmdStr := maskSensitiveData(cmd.String())
-		log.Infof("Running Helm command: %s", cmdStr)
+		log.Infof("Running Helm command for operation %s: %s", cmdStr)
```
Comment (mgianatagh) [inline pkg/helm/helmcli.go:172]: Looks like an extra %s is in the format string

Comment (pfmackin) [reply pkg/helm/helmcli.go]: Fixed, I also changed not to display cmdstr unless there is an error

**Review by mgianatagh — DISMISSED** [review]

**Review by mgianatagh — DISMISSED** [review]

File: pkg/log/vzlog/vzlog.go [hunk 2]
```diff
@@ -236,12 +309,17 @@ func (v *verrazzanoLogger) SetZapLogger(zap *zap.SugaredLogger) {
 	v.sLogger = zap
 }
 
-// GetZapLogger gets the zap logger
+// Clone zap logger gets a clone of the zap logger
```
Comment (kminder) [inline pkg/log/vzlog/vzlog.go:312]: Nit.  The original comment was godoc compliant, right?

Comment (pfmackin) [reply pkg/log/vzlog/vzlog.go:312]: will fix next commit

File: platform-operator/controllers/verrazzano/component/spi/component.go [hunk 3]
```diff
@@ -24,8 +24,8 @@ type ComponentContext interface {
 	IsDryRun() bool
 	// Copy returns a copy of the current context
 	Copy() ComponentContext
-	// For returns a copy of the current context with an updated logging component field
-	For(comp string) ComponentContext
+	// Init returns a copy of the current context with an updated logging component field
+	Init(comp string) ComponentContext
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/component/spi/component.go:28]: Nit but For seemed better to me.

File: platform-operator/controllers/verrazzano/component/spi/testdata/basicDevMerged.yaml [hunk 4]
```diff
@@ -6,7 +6,7 @@ spec:
   profile: "dev"
   environmentName: default
   defaultVolumeSource:
-    emptyDir: {}
+    emptyDir: { }
```
Comment (kminder) [inline platform-operator/controllers/verrazzano/component/spi/testdata/basicDevMerged.yaml:9]: Why change this?

**Review by kminder — APPROVED** [review]

**Pull request merged**